lms/services/application_instance.py

```diff
@@ -0,0 +1,54 @@
+from typing import Optional
+
+from sqlalchemy import or_
+
+from lms.models import ApplicationInstance
+
+
+class ApplicationInstanceService:
+    def __init__(self, db):
+        self._db = db
+
+    def find(self, query: str) -> Optional[ApplicationInstance]:
+        """Query an ApplicationInstance by both the unique pk and consumer_key."""
+        return (
+            self._db.query(ApplicationInstance)
+            .filter(
+                or_(
+                    ApplicationInstance.id == query if query.isdigit() else None,
+                    ApplicationInstance.consumer_key == query,
+                )
+            )
+            .one_or_none()
+        )
+
+    # pylint: disable=invalid-name
+    def get(self, pk: int) -> Optional[ApplicationInstance]:
+        """Query an ApplicationInstance by primary key."""
+
+        return self._db.query(ApplicationInstance).get(pk)
+
+    @staticmethod
+    def update_settings(
```

Comment:
I did this because I didn't wan the caller needing to know where to store the stuff "canvas" -> "group_enabled" vs just passing a boolean here